discovery_engine/test/integration/source_preferences_test.dart

```diff
@@ -53,16 +54,29 @@ void main() {
     test('addSourceToExcludedList adds excluded source', () async {
       engine = await initEngine(data, server.port);
```

Comment:
:wrench: the engine is initialized twice, once during `setUp()` and once here. this second call can be removed as the `setUp()` is run before every test anyways. same for the other tests of this module.
```suggestion
```